cloudinit/sources/DataSourceAeza.py

```diff
@@ -0,0 +1,78 @@
+# Copyright (C) 2024 Aeza.net.
+#
+# Author: Egor Ternovoy <cofob@riseup.net>
+#
+# This file is part of cloud-init. See LICENSE file for license information.
+
+import logging
+
+from cloudinit import sources, dmi, util
+
+LOG = logging.getLogger(__name__)
+
+BUILTIN_DS_CONFIG = {
+    "metadata_url": "http://77.221.156.49/v1/cloudinit/{id}/",
+}
+
+
+class DataSourceAeza(sources.DataSource):
+
+    dsname = "Aeza"
+
+    def __init__(self, sys_cfg, distro, paths, ud_proc=None):
+        super().__init__(sys_cfg, distro, paths, ud_proc)
+
+        self.ds_cfg = util.mergemanydict([self.ds_cfg, BUILTIN_DS_CONFIG])
+
+        url_params = self.get_url_params()
+        self.timeout_seconds = url_params.timeout_seconds
+        self.max_wait_seconds = url_params.max_wait_seconds
+        self.retries = url_params.num_retries
+        self.sec_between_retries = url_params.sec_between_retries
+
+        system_uuid = dmi.read_dmi_data("system-uuid")
+        self.metadata_address = (
+            self.ds_cfg["metadata_url"].format(
+                id=system_uuid,
+            )
+            + "%s"
+        )
+
+    @staticmethod
+    def ds_detect():
+        return dmi.read_dmi_data("system-manufacturer") == "Aeza"
+
+    def _get_data(self):
+        md, ud, vd = util.read_seeded(
```

Comment:
A commit commit landed in tip of main that is incompatible with this changeset as read_seeded now returns a 4-tuple from util.read_seeded which looks for a supplemental/optional network-config file.We need to git rebase on this branch to refresh to latest commit on main and possibly ignore that network-config 4-tuple return value. I haven't rebased your PR against upstream/main and pushed my suggested changes as a commit because we may actually want to think about providing a supplemental parameter to allow read_seeded to ignore GETs of network-config since network-config isdisregared by this platform and we don't want to waste URL retries on an unexpected/irrelevant file.